rotkehlchen/tests/api/test_assets.py

```diff
@@ -499,6 +499,51 @@ def test_get_all_assets(rotkehlchen_api_server):
         assert 'underlying_tokens' in entry
         assert entry['evm_chain'] in [x.to_name() for x in ChainID]
 
+    # test that wrong combination of evm_chain & asset_type fails.
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'allassetsresource',
+        ),
+        json={'asset_type': 'fiat', 'evm_chain': 'ethereum'},
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='Filtering by evm_chain is only supported by evm tokens',
+        status_code=HTTPStatus.BAD_REQUEST,
+    )
+
+    # check that filtering by evm_chain works.
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'allassetsresource',
+        ),
+        json={'evm_chain': 'ethereum', 'limit': 10, 'offset': 0},
```

Comment:
can you filter by symbol and `evm_chain` removing the limit? As it is now could be that you are returning the results with chain ethereum first and there are more that due the limit are not shown